### PR TITLE
Rewrite all js files into ts files

### PR DESCRIPTION
**Pull request by ChAoSUnItY**
Actually we can use .d.ts to save our life, but after a lookup, there're several type bugs cannot be done within just .d.ts files, so this refactor to ensure all structures and resolver can have correct type declarations.

### CODE REVIEW

File: src/DataResolver.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+     */
+    public resolveUserID(user: WikidotUser | string | SiteMember| WikidotPage): string | undefined {
+        switch (user.constructor) {
+            case WikidotUser || SiteMember:
+                return (user as WikidotUser).id.toString();
+            case String:
+                return user as string || undefined;
+            case WikidotPage:
+                let wp = user as WikidotPage;
+                if (wp.authors.size) return wp.authors.values().next().value;
```
Comment (Zokhoi) [inline src/DataResolver.ts]: The returned value here and the line below should be the user id, not the `SiteMember` object.

File: src/structures/SiteMember.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+     * @type {Number}
+     */
+    public highestRating: number;
+    /**
+     * The total rating this member has gotten for all owned pages on the site
+     * @type {Number}
+     */
+    public totalRating: number;
+
+    constructor(client: Scpper, site: SiteResolvable, data: object) {
```
Comment (Zokhoi) [inline src/structures/SiteMember.ts]: Parameter `site` should be allowed direct passing of type `string`.

**Review by Zokhoi — CHANGES_REQUESTED** [review]
The PR is pretty much fine, just some minor adjustments needed. Please change them and I'll merge the PR.

File: src/DataResolver.ts (hunk 1 above)
Comment (ChAoSUnItY) [reply src/DataResolver.ts]: Replaced with 
```ts
value.id.toString();
```

File: src/structures/SiteMember.ts (hunk 2 above)
Comment (ChAoSUnItY) [reply src/structures/SiteMember.ts]: Changed into
```ts
site: SiteResolvable | string
```

**Pull request merged**